### ISSUE TITLE
Create use-in-view-ref hook

### ISSUE THREAD

**Issue by imbhargav5** [imbhargav5/rooks#365]
Let's create a simple hook like `useInViewRef` using intersection observer. Some API examples should already be there. We can do something like.

```jsx
const [ref, isInView] = useInView(options)
```

What do you think? @maciekgrzybek

**Comment by maciekgrzybek** [comment]
Yeah I'm happy to do it, I did something similar for svelte https://www.npmjs.com/package/svelte-inview

**Comment by imbhargav5** [comment]
@maciekgrzybek Let's focus on this one first. Meanwhile I am trying to make a list of the new hooks to make and will take up one of them.

**Comment by maciekgrzybek** [comment]
OK cool I'll try to do something this weekend ')

**Comment by maciekgrzybek** [comment]
Ok I got the hook I just need to make sure it all works fine. Can you tell me what's the best to test new hooks in this repo?

**Comment by maciekgrzybek** [comment]
And I mean manual testing, not the automated ones :)

**Comment by maciekgrzybek** [comment]
PR ready for review @imbhargav5  https://github.com/imbhargav5/rooks/pull/374
